### PR TITLE
Add plotting module and advanced documentation

### LINKED ISSUE #251: ENH: Allow passing multiple phases to potential references
A potential reference connected to a bus can be set on any (one) phase of the bus, in this case, the potential on that phase is fixed at 0V. Another type of potential reference exist for buses without neutral that set the sum of potentials of the bus to 0V.

We cannot currently use the second type with a bus that has a neutral. We cannot also choose what phases are considered for the second type. I suggest to change the `PotentialRef` constructor to allow the user to specify any phase configuration of the bus to be used as a reference of potentials.

In the docs, we should encourage the users to not pass the phases and rely on the default behavior as it always chooses the best configuration for any given scenario. In an "Advanced" section (to be created) in the docs, we can add a page that explains the advanced usages of the potential reference and show, using plots, the difference in the results.

### PR DESCRIPTION
**Pull request by alihamdan**
Closes #251

### CODE REVIEW

File: doc/_static/css/custom.css [hunk 1]
```diff
@@ -77,3 +77,14 @@ h6,
   font-weight: unset;
   line-height: unset;
 }
+
+/*Roseau colors */
```
Comment (alihamdan) [inline doc/_static/css/custom.css:81]: With these css properties, we can now reference and use the colors in our documentation

File: doc/index.md [hunk 2]
```diff
@@ -38,10 +38,22 @@ and newer and can be installed with:
 ```{toctree}
 ---
 maxdepth: 2
+caption: Installation and License
 ---
 Installation
 ```
 
+## License
```
Comment (alihamdan) [inline doc/index.md:46]: Moved the license to the top as it was not very visible before and needing a scroll

File: roseau/load_flow/network.py [hunk 3]
```diff
@@ -312,14 +312,15 @@ def lines_frame(self) -> gpd.GeoDataFrame:
                     line.bus1.id,
                     line.bus2.id,
                     line.parameters.id,
+                    line.length.m,
```
Comment (alihamdan) [inline roseau/load_flow/network.py:315]: Added the length to the dataframe. It is useful information and it also useful as part of the tooltip of the interactive map.

File: roseau/load_flow/plotting.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    ax.arrow(potential2.real, potential2.imag, voltage.real, voltage.imag, color=color)  # type: ignore
+    rot = _get_rot(voltage)
+    ha, va = _get_align(rot)
+    ax.annotate(f"{abs(voltage):.0f}V", (midpoint.real, midpoint.imag), ha=ha, va=va, rotation=rot)  # type: ignore
+
+
+#
+# Phasor plotting functions
+#
+def plot_voltage_phasors(
```
Comment (alihamdan) [inline roseau/load_flow/plotting.py:79]: In the future I am gonna factorize this and add `plot_current_phasors` function as well but this is a good place to start for now.

File: roseau/load_flow/units.py [hunk 5]
```diff
@@ -41,10 +45,61 @@
...
+    Array: TypeAlias = np.ndarray[Any, Any]
+    UnitLike = str | dict[str, Scalar] | UnitsContainer | Unit
+
+    NpNumT = TypeVar("NpNumT", bound=np.number[Any])
+    MagBound = Scalar | Array | Sequence[Scalar | Array] | Sequence[Sequence[Scalar | Array]]
+    MagT = TypeVar("MagT", bound=MagBound)
+    MagT_co = TypeVar("MagT_co", covariant=True, bound=MagBound)
+
+    # Redefine Q_ with support for complex and better type hints
+    class Q_(NumpyQuantity[MagT_co]):  # type: ignore # noqa: N801
```
Comment (alihamdan) [inline roseau/load_flow/units.py:59]: This is done to add complex number support to the quantity class and to make the typing compatible with both VSCode and Pycharm (I tested both)

File: scripts/plot_network_catalogue.py [hunk 6]
```diff
@@ -170,7 +170,7 @@ def prettify_network_name(name: str) -> str:
         # Create the map
         zoom_start = 12 if network_name.startswith("MV") else 16
         title = prettify_network_name(name=network_name)
-        m = RoseauMap(location=list(reversed(buses_gdf.unary_union.centroid.coords[0])), zoom_start=zoom_start)
+        m = RoseauMap(location=list(reversed(buses_gdf.union_all().centroid.coords[0])), zoom_start=zoom_start)
```
Comment (alihamdan) [inline scripts/plot_network_catalogue.py:173]: @benoit9126 `GeoSeries.unary_union` started emitting a deprecation warning. You may want to change it all over our code base when you see it used.

**Review by benoit9126 — APPROVED** [review]

**Pull request merged**